crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1995,14 +1996,52 @@ fn run_rustfmt(
             cmd
         }
         RustfmtConfig::CustomCommand { command, args } => {
-            let mut cmd = process::Command::new(command);
+            let cmd = PathBuf::from(&command);
+            let mut components = cmd.components();
+
+            // to support rustc's suggested, default configuration
+            let mut cmd = match components.next() {
+                Some(std::path::Component::CurDir) => {
```

Comment:
This should probably work with `build/host/rustfmt` as well; special-casing the leading `./` sounds very fragile.

Reply:
Ye, we can just [Path::join](https://doc.rust-lang.org/std/path/struct.Path.html#method.join) the rustfmt path onto the base here. If the path is absolute it'll replace the base, otherwise it"ll append which is exactly the semantics we want to have here.